includes/class-wc-referralcandy-integration.php

```diff
@@ -73,67 +87,73 @@ public function sanitize_settings($settings) {
             return $settings;
         }
 
-        public function is_option_enabled($option_name) {
+        private function is_option_enabled($option_name) {
             return $this->get_option($option_name) == 'yes'? true : false;
         }
 
-        public function woocommerce_order_referralcandy($order_id) {
-            $wc_pre_30 = version_compare( WC_VERSION, '3.0.0', '<');
+        public function add_referralcandy_data($post_id) {
+            try {
+                if (get_post($post_id)->post_type == 'shop_order') {
+                    $wc_order = new WC_Order($post_id);
 
-            $order = new WC_Order($order_id);
+                    // save meta datas for later use
+                    $wc_order->update_meta_data('api_id',       $this->api_id);
+                    $wc_order->update_meta_data('secret_key',   $this->secret_key);
+                    $wc_order->update_meta_data('browser_ip',   $_SERVER['REMOTE_ADDR']);
+                    $wc_order->update_meta_data('user_agent',   $_SERVER['HTTP_USER_AGENT']);
 
-            // https://en.support.wordpress.com/settings/general-settings/2/#timezone
-            // This option is set when a timezone name is selected
-            $timezone_string = get_option('timezone_string');
+                    // prevent admin cookies from automatically adding a referrer_id; this can be done manually though
+                    if (is_admin() == false) {
+                        $wc_order->update_meta_data('referrer_id',  $_COOKIE['rc_referrer_id']);
+                    }
```

Comment:
Is the idea here that we're trying to prevent referrals by the retailer? What if they're trying to test referrals?

Reply:
yeap! they can still create test purchases via store purchases in incognito mode, or they could add the AIC parameter themselves when creating / updating the orders :)